pf/internal/schemashim/block_schema.go

```diff
@@ -76,30 +73,27 @@ func (s *blockSchema) Elem() interface{} {
 	if _, ok := s.block.Type().(basetypes.ListTypable); ok {
 		if twet, ok := s.block.Type().(attr.TypeWithElementType); ok {
 			r, ok := asObjectType(twet.ElementType())
-			if !ok {
-				panic(fmt.Errorf("List-nested block expect an ObjectTypeable "+
-					"block.Type().ElemType, but got %v", s.block.Type()))
-			}
+			contract.Assertf(ok, "List-nested block expect an ObjectTypeable "+
+				"block.Type().ElemType, but got %v", s.block.Type())
 			return r
 		}
-		panic(fmt.Errorf("List-nested block has a ListTypeable type that does not implement "+
-			"TypeWithElementType: %v", s.block.Type()))
+		contract.Assertf(false, "List-nested block has a ListTypeable type that does not implement "+
```

Comment:
I believe we have `contract.Failef`
```suggestion
		contract.Failf("List-nested block has a ListTypeable type that does not implement "+
```